src/Symfony/Component/Translation/Extractor/PhpExtractor.php

```diff
@@ -307,7 +349,32 @@ protected function parseTokens(array $tokens, MessageCatalogue $catalog, string
                     $metadata = $catalog->getMetadata($message, $domain) ?? [];
                     $normalizedFilename = preg_replace('{[\\\\/]+}', '/', $filename);
                     $metadata['sources'][] = $normalizedFilename.':'.$tokens[$key][2];
+
+                    if (\count($variables) > 0) {
+                        $variablesNote = [
+                                'category' => MessageCatalogue::METADATA_AVAILABLE_VARIABLES_KEY,
+                                'content' => MessageCatalogue::METADATA_AVAILABLE_VARIABLES_PREFIX.implode(', ', $variables),
+                        ];
+
+                        // Update old variables note (if any)
+                        if (isset($metadata['notes'])) {
+                            foreach ($metadata['notes'] as $index => $note) {
+                                if (isset($note['category']) && MessageCatalogue::METADATA_AVAILABLE_VARIABLES_KEY === $note['category']) {
+                                    // Keep the highest variables count
+                                    if (\count($variables) > substr_count($note['content'], ',')) {
+                                        $metadata['notes'][$index] = $variablesNote;
+                                    }
+
+                                    break;
+                                }
+                            }
+                        } else {
+                            $metadata['notes'][] = $variablesNote;
+                        }
+                    }
```

Comment:
@liarco this code is repeated several times, shouldn't we factorize it in a separate method, in a new file?